index.js

```diff
@@ -59,6 +59,13 @@ var fsReadFile = Promise.promisify(fs.readFile, {context: fs});
 var fsStat = Promise.promisify(fs.stat, {context: fs});
 var fsWriteFile = Promise.promisify(fs.writeFile, {context: fs});
 
+function md5(file) {
+  return fsReadFile(file, 'utf8').then(function(contents) {
```

Comment:
We should use webpack's inputFileSystem to read these files. webpack soft caches the files read that way. Get the file system `compilation.inputFileSystem`.

`fsReadFile` here is just used to read the time stamp. And the FileSerializer uses it for the cache. Nothing that gets read multiple times like the files that need to be md5'd.